frontend-auth-module/projects/sl-auth-lib/src/lib/components/otp/otp.component.ts

```diff
@@ -0,0 +1,154 @@
+import { NgxOtpInputConfig } from "ngx-otp-input";
+import { EndpointService } from '../../services/endpoint/endpoint.service';
+import { ApiBaseService } from '../../services/base-api/api-base.service';
+import { Router } from '@angular/router';
+import { Location } from '@angular/common';
+import { StateService } from '../../services/state/state.service';
+import { Subscription, interval } from 'rxjs';
+
+@Component({
+  selector: 'lib-otp',
+  templateUrl: './otp.component.html',
+  styleUrl: './otp.component.css'
+})
+export class OtpComponent {
+  configData: any;
+  endPointService: EndpointService;
+  location: Location;
+  otpInput: boolean = true;
+  regFormData: any;
+  otp!: any;
+  checkbox:boolean = false;
+
+  baseApiService: ApiBaseService;
+  router: Router;
+  timeLeft: number = 60;
+
+  constructor(private stateService: StateService) {
+    this.baseApiService = inject(ApiBaseService);
+    this.endPointService = inject(EndpointService);
+    this.router = inject(Router);
+    this.location = inject(Location);
+  }
+
+  ngOnInit() {
+    this.startTimer();
+    this.regFormData = this.stateService.getData();
+    if (this.regFormData) {
+      console.log('Received state data:', this.regFormData);
+    } else {
+      console.error('No state data found');
+    }
+    this.fetchConfigData();
+  }
+
+  async fetchConfigData() {
+    try {
+      this.configData = await this.endPointService.getEndpoint();
+    } catch (error) {
+      console.error("An error occurred while fetching configData:", error);
+    }
+  }
+
+  otpInputConfig: NgxOtpInputConfig = {
+    otpLength: 6,
+    autofocus: true,
+    // classList: {
+    //   inputBox: "my-super-box-class",
+    //   input: "my-super-class",
+    //   inputFilled: "my-super-filled-class",
+    //   inputDisabled: "my-super-disable-class",
+    //   inputSuccess: "my-super-success-class",
+    //   inputError: "my-super-error-class"
+    // }
+  };
+
+  handleFillEvent(value: string): void {
+    this.otp = value;
+  }
+
+  navigateBack() {
+    this.location.back();
+  }
+
+  generateOTP() {
+    console.log(this.regFormData)
+    let payload = {
+      email: this.regFormData?.email,
+      password: this.regFormData?.password,
+      name: this.regFormData?.name,
+    }
+
+    let selectedApiPath: string;
+    if (this.regFormData.fromPage === "signup") {
+      selectedApiPath = "otpGenerationApiPathForRegistration"
+    } else if (this.regFormData.fromPage === "reset") {
+      selectedApiPath = "otpGenerationApiPathForResetPassword"
+      delete payload.name;
+    }
+    else {
+      selectedApiPath = "";
+    }
+
+    this.baseApiService
+      .post(
+        this.configData?.baseUrl, this.configData?.[selectedApiPath], payload)
+      .subscribe((res: any) => {
+        this.otpInput = true;
+        alert(res?.message);
+        this.startTimer();
+      },
+        (err: any) => {
+          alert(err?.error?.message);
+        }
+      );
+  }
+
+
+  verifyOTP() {
```

Comment:
generateOTP and verifyOTP have similar lines of code, generalize the functions into a single function and write conditions specific to generate and verify only where its applicable